src/OpenLoco/src/GameCommands/Track/RemoveTrainStation.cpp

```diff
@@ -0,0 +1,141 @@
+        if (initialElTrack == nullptr)
+        {
+            return FAILURE;
+        }
+
+        if (!sub_431E6A(initialElTrack->owner(), reinterpret_cast<const World::TileElement*>(initialElTrack)))
+        {
+            return FAILURE;
+        }
+
+        const auto& trackPieces = World::TrackData::getTrackPiece(args.trackId);
+        const auto& argPiece = trackPieces[args.index];
+        const auto trackStart = args.pos - World::Pos3(Math::Vector::rotate(World::Pos2(argPiece.x, argPiece.y), args.rotation), argPiece.z);
+
+        // sub dx, [ebp+edi+5] ???
+
+        StationId foundStationId = StationId::null;
+        currency32_t totalCost = 0;
+
+        for (auto& piece : trackPieces)
+        {
+            const auto trackLoc = trackStart + World::Pos3{ Math::Vector::rotate(World::Pos2{ piece.x, piece.y }, args.rotation), piece.z };
+
+            auto* elTrack = getElTrack(trackLoc, args.rotation, args.type, args.trackId, piece.index);
+
+            // add dx, [ebp+edi+5] ???
+
+            if (elTrack == nullptr)
+            {
+                if (_112C7A9 && (flags & Flags::apply) != 0)
+                {
+                    auto* station = StationManager::get(foundStationId);
+                    removeTileFromStation(foundStationId, trackStart, args.rotation);
+                    station->invalidate();
+
+                    recalculateStationCenter(foundStationId);
+                    recalculateStationModes(foundStationId);
+                    station->updateLabel();
+                    station->invalidate();
+
+                    sub_48D794(*station);
+                }
+                break;
+            }
+            else
+            {
+                auto* nextEl = elTrack->next();
+                auto* stationEl = nextEl->as<World::StationElement>();
+                if (stationEl == nullptr)
+                    return FAILURE;
+
+                if (stationEl->isAiAllocated())
+                    _112C7A9 = false;
+
+                foundStationId = stationEl->stationId();
+
+                auto* stationObj = ObjectManager::get<TrainStationObject>(stationEl->objectId());
+                auto removeCostBase = Economy::getInflationAdjustedCost(stationObj->sellCostFactor, stationObj->costIndex, 8);
+                const auto cost = (removeCostBase * World::TrackData::getTrackMiscData(args.trackId).costFactor) / 256;
+                totalCost += cost;
+
+                if ((flags & Flags::apply) != 0)
+                {
+                    elTrack->setClearZ(elTrack->clearZ() - stationObj->height);
+                    Ui::ViewportManager::invalidate(World::Pos2(args.pos), stationEl->baseZ(), stationEl->clearZ(), ZoomLevel::eighth);
+                    elTrack->setHasStationElement(false);
+                    World::TileManager::removeElement(*nextEl);
+                }
+
+                // ebp += 0x10 ??
```

Comment:
Not sure how to translate this to the current loop (cf. 0x0048C5D6).

@duncanspumpkin any advice?

Reply:
It's for the track element iterator, not the tile element loop.